### ISSUE TITLE
contentRouting.findProviders(cid) fails when provider peer is not stored locally

### ISSUE THREAD

**Issue by dejano-with-tie** [libp2p/js-libp2p#889]
- **Version**: 0.30.8
- **Platform**: Linux desktop 5.10.13-arch1-2
- **Subsystem**: Content Routing (DHT)

#### Description:

When dht does not have provider peer stored locally and it is retrieved from the network, the following call
```Javascript
libp2p.contentRouting.findProviders(cid)
```
will throw an error:
```
libp2p-example/node_modules/libp2p/src/peer-store/address-book.js:294
      throw errcode(new Error('multiaddrs must be provided'), ERR_INVALID_PARAMETERS)
                    ^
Error: multiaddrs must be provided
    at AddressBook._toAddresses (libp2p-example/node_modules/libp2p/src/peer-store/address-book.js:294:21)
    at AddressBook.add (libp2p-example/node_modules/libp2p/src/peer-store/address-book.js:230:28)
    at libp2p-example/node_modules/libp2p/src/content-routing/utils.js:22:27
    at map (libp2p-example/node_modules/it-map/index.js:14:11)
    at async filter (libp2p-example/node_modules/it-filter/index.js:11:20)
    at async requirePeers (libp2p-example/node_modules/libp2p/src/content-routing/utils.js:58:20)
    at async ContentRouting.findProviders (libp2p-example/node_modules/libp2p/src/content-routing/index.js:57:5)
    at async all (libp2p-example/node_modules/it-all/index.js:13:20)
    at async libp2p-example/find-providers.bug.js:53:23 {
  code: 'ERR_INVALID_PARAMETERS'
}
```
`findProviders` is calling `storeAddresses` with found provider peer. The problem is that only `peer.peerId` has value set but not the `peer.multiaddrs` and `storeAddresses` call is failing because of undefined addresses.

#### Steps to reproduce the error:
* node_1 provides _cid_
* node_2 joins the network
* node_2 asks who has _cid_ -> crash

The error can be reproduced with the following snippet:
```Javascript
'use strict'

const Libp2p = require('libp2p/');
const TCP = require('libp2p-tcp');
const Mplex = require('libp2p-mplex');
const {NOISE} = require('libp2p-noise');
const KadDHT = require('libp2p-kad-dht');
const CID = require('cids');

const delay = require('delay');
const all = require('it-all');

const createNode = async () => {
    const node = await Libp2p.create({
        addresses: {
            listen: ['/ip4/0.0.0.0/tcp/0']
        },
        modules: {
            transport: [TCP],
            streamMuxer: [Mplex],
            connEncryption: [NOISE],
            dht: KadDHT
        },
        config: {
            dht: {
                enabled: true
            }
        }
    });

    await node.start();
    return node;
};

(async () => {
    const node1 = await createNode();

    // Node 1 provides
    const cid = new CID('bagaaiera3jkkvjplzpodkgpggnkqr7anicgpxzfkdevgtfuz7d7jjvgyejfq');
    await node1.contentRouting.provide(cid);
    console.log('Node %s is providing %s', node1.peerId.toB58String(), cid.toBaseEncodedString());
    await delay(200);

    // Node 2 joins
    const node2 = await createNode();

    node2.peerStore.addressBook.set(node1.peerId, node1.multiaddrs);
    await node2.dial(node1.peerId);

    // wait for propagation
    await delay(200);

    // who has cid? this fails
    const providers = await all(node2.contentRouting.findProviders(cid, {timeout: 3000}));
    console.log('Found provider:', providers[0].id.toB58String());
})();
```

**Comment by dejano-with-tie** [comment]
I found out that this is fixed with https://github.com/libp2p/js-libp2p-kad-dht/commit/71952820ef3f737204b7a615db69ae680ef652a8#diff-38171968c0a0b8223fb6df428b7553271a854160e8db4c112a49ac28a5a256d0R153

It will work once dht package is released I guess.

**Comment by vasco-santos** [comment]
@dejano-with-tie yes, I am testing it and I will release it.

The multiaddr was not being added but now we spread all the properties

**Comment by vasco-santos** [comment]
Just shipped `0.21.0` and tried it out. It fixes the issue

Thanks for reporting @dejano-with-tie
